tests/unit/test_useragent.py

```diff
@@ -193,3 +198,57 @@ def test_from_environment_unknown_platform(monkeypatch):
     monkeypatch.setattr(platform, 'release', lambda: '0.0.1')
     uas = UserAgentString.from_environment()
     assert ' os/other md/FooOS#0.0.1 ' in uas.to_string()
+
+
+def test_user_agent_string_with_registered_features(client_context):
+    uas = UserAgentString.from_environment()
+    uas.set_client_features({'A'})
+    register_feature_id('WAITER')
+
+    uafields = uas.to_string().split(' ')
+    feature_field = [field for field in uafields if field.startswith('m/')][0]
+    feature_list = feature_field[2:].split(',')
+    assert sorted(feature_list) == ['A', 'B']
+
+
+def test_register_feature_id(client_context):
+    register_feature_id('WAITER')
+    ctx = get_context()
+    assert ctx.features == {'B'}
+
+
+def test_register_unknown_feature_id_raises(client_context):
+    with pytest.raises(ValueError) as excinfo:
+        register_feature_id('MY_FEATURE')
+    assert "Unknown feature id" in str(excinfo.value)
```

Comment:
Is there a world where someone is on a newer downstream package, say `s3transfer`, and it registers a feature that isn't in botocore's mapping? 

Wondering if it's better to fail more gracefully? Thinking about the mismatch issues we had with the S3 checksums rollout.

Otherwise we have to update the minimum pin each time we add a new feature, right?

Reply:
Addressed in https://github.com/boto/botocore/pull/3389/commits/6eec8771bf168ed82df2ac9845edca2004a1100b